packages/components/src/typography/typography.tsx

```diff
@@ -0,0 +1,169 @@
+
+type TypographyElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "span" | "p";
+
+type TypographySize =
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "body"
+  | "sm"
+  | "xs"
+  | "button"
+  | "caption"
+  | "overline";
+
+type TypographyColor =
+  | "alert"
+  | "base"
+  | "brand"
+  | "info"
+  | "success"
+  | "warning"
+  | "white";
+
+function elementFromSize(size: TypographySize): TypographyElement {
+  switch (size) {
+    case "h1":
+    case "h2":
+    case "h3":
+    case "h4":
+    case "h5":
+      return size;
+    default:
+      return "p";
+  }
+}
+
+function styleFromColor(color: TypographyColor): TwStyle {
+  switch (color) {
+    case "alert":
+      return tw`text-alert-700`;
+    case "brand":
+      return tw`text-brand-700`;
+    case "info":
+      return tw`text-info-700`;
+    case "success":
+      return tw`text-success-700`;
+    case "warning":
+      return tw`text-warning-700`;
+    case "white":
+      return tw`text-white`;
+    default:
+      return tw`text-font-base`;
+  }
+}
+interface BaseTypographyProps {
+  bold: boolean;
+  color: TypographyColor;
+}
+
+const Base = styled.span<BaseTypographyProps>(({ bold, color }) => [
+  tw`font-sans`,
+  bold ? tw`font-bold` : tw`font-normal`,
+  styleFromColor(color),
+]);
+const H1 = tw(Base)`text-h1 leading-h1 capitalize`;
+const H2 = tw(Base)`text-h2 leading-h2 capitalize`;
+const H3 = tw(Base)`text-h3 leading-h3 capitalize`;
+const H4 = tw(Base)`text-h4 leading-h4 capitalize`;
+const H5 = tw(Base)`text-h5 leading-body capitalize`;
+const Body = tw(Base)`text-body leading-body`;
+const BodySm = tw(Base)`text-sm leading-sm`;
+const BodyXs = tw(Base)`text-xs leading-xs`;
+const Button = tw(Base)`text-sm leading-body`;
+const Caption = tw(Base)`text-xs leading-sm`;
+const Overline = tw(Base)`text-xs leading-sm uppercase`;
+
+function styledComponentFromSize(size: TypographySize): typeof Base {
+  switch (size) {
+    case "h1":
+      return H1;
+    case "h2":
+      return H2;
+    case "h3":
+      return H3;
+    case "h4":
+      return H4;
+    case "h5":
+      return H5;
+    case "body":
+      return Body;
+    case "sm":
+      return BodySm;
+    case "xs":
+      return BodyXs;
+    case "button":
+      return Button;
+    case "caption":
+      return Caption;
+    case "overline":
+      return Overline;
+    default:
+      throw new TypeError(`Received invalid size option '${size}'`);
+  }
+}
+
+type TypographyProps = {
+  /**
+   * This prop can be used to specify which size text element should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
+   */
+  as?: TypographyElement;
+  /**
+   * Specifies font weight as either bold or normal (defaults to normal).
+   */
+  bold?: boolean;
```

Comment:
Per Sean, originally all headers were only going to be bold. But designers said they needed non-bold options, so now most sizes have a bold and non-bold design in Figma. We don't have strict usages tied to each one yet, so I think we should leave this as a prop for now